src/query/block/meta.go

```diff
@@ -158,6 +161,31 @@ func (m ResultMetadata) IsDefault() bool {
 	return m.Exhaustive && m.LocalOnly && len(m.Warnings) == 0
 }
 
+// VerifyTemporalRange will verify that each resolution seen is below the
+// given step size, adding warning headers if it is not.
+func (m *ResultMetadata) VerifyTemporalRange(step time.Duration) {
+	// NB: this map is unlikely to have more than 2 elements in real execution,
+	// since these correspond to namespace count.
+	invalidResolutions := make(map[time.Duration]struct{}, 10)
```

Comment:
Maybe a silly question, but why not use `make(map[time.Duration]struct{}, 0, 10)` here so that you can append below? This way you won't have to worry about indexing out of bounds (even if it is unlikely)

Reply:
It's a map not a slice :p 

Realistically not expecting this to have more than 2 or at most 3 different values, which is why I'm only initing to 10 here :+1:

Reply:
lol, missed that! My bad. SGTM